README.md

```diff
@@ -98,4 +100,41 @@ We can extend the concept of the `as` syntax by generalizing the right hand side
 import value from "module" as {key1: "value1", key2: [1, 2, 3]};
 ```
 
-This would allow module attributes to scale up to multiple pieces of data, not privileging a single attribute.
+This would allow module attributes to scale to support a variety of metadata. It is currently unclear if the generalized constant form will be pursued in this proposal or in a follow up. In either case the addition of the generalized constant form is not hard to implement on top of the existing [spec text](https://tc39.github.io/proposal-module-attributes/). 
+
+## FAQ
+
+### Why not out of band?
+
+Why not both? The champions of this proposal think that exploring both an in- and out of band solutions to this problem are desirable. Depending on the environment in which JavaScript is being executed in either approach could be seen as desirable.
+
+While an in band solution is more verbose, it is also more straightforward for developers to adopt. For large project with many dependencies developers will not have to worry about creating a large manifest by compiling the meta data of all of their dependencies. Module authors will also not have to worry about shipping manifest in order for consumers to be able to run their modules.
+
+The [import-maps proposal][] is a great example of an out of band manifest that could be created at installation time, but it's focus is on modules and at this time all necessary information to generate an applications import map could be found in the package.json of it's dependencies.
+
+While there very well could be a future where an out of band manifest could be used for the purpose of identifying module type, the amount of infrastructure necessary for ecosystem adoption is significantly higher than the in band solution proposed here.
+
+### Are there cross-environment concerns?
+
+Absolutely. This proposal is not attempting to standardize what `types` hosts will be allowed to support. It will be up to each runtime to decide which `types` to support, as well as how to handle the the case of an unsupported module `type` being imported.
+
+Module specifiers are an example of prior art for this type of decision making. It is left up to the host how to resolve a module specifier into a URL that will eventually be loaded. The [import-maps proposal][] is a great example of work done outside of TC39 to allow hosts to share behavior for undefined parts of the spec.
```

Comment:
I find citing import-maps here a bit odd. import-maps's specification is about as cross-environment as HTML's module loading sections, and both have been implemented cross-environment anyway.

Reply:
Just dropped the reference for now, we can revisit